### PR TITLE
Refactor right panel header buttons

### PR DESCRIPTION
**Pull request by lukebarnard1**
To reduce repetition of code, simplify the process of adding header buttons and remove the need for many handler functions.

### CODE REVIEW

**Comment by lukebarnard1** [comment]
(I'll squash merge this)

File: src/components/structures/RightPanel.js [hunk 1]
```diff
@@ -17,17 +17,65 @@ limitations under the License.
...
+            { isHighlighted ? <div className="mx_RightPanel_headerButton_highlight"></div> : <div/> }
+        </AccessibleButton>;
+    }
+}
+
+HeaderButton.propTypes = {
+    // If currentPhase is one of the specified phases, the button will be highlighted
+    phases: PropTypes.arrayOf(PropTypes.string).isRequired,
+    // The currentPhase of the RightPanel
+    currentPhase: PropTypes.string.isRequired,
```
Comment (dbkr) [inline src/components/structures/RightPanel.js]: Is there likely to be any need for the button to know the actual phases it represents or what the current phase is? Could we just pass in whether it should be highlighted or not?

Comment (lukebarnard1) [reply src/components/structures/RightPanel.js]: >Could we just pass in whether it should be highlighted or not?

I was hoping to abstract this logic away such that, for example, we can't get two buttons highlighted at once. It's _probably_ overkill but if every single header button is going to embody this logic, it seems inefficient to dupe it for each one.

Comment (lukebarnard1) [reply src/components/structures/RightPanel.js]: Also, it needs to know for the dispatch so :man_shrugging:

**Pull request merged**